### PR TITLE
append more logs in attach detach scripts

### PR DESCRIPTION
**Pull request by SharpRazor**
Signed-off-by: SharpRazor <bjcb@cn.ibm.com>

### CODE REVIEW

**Comment by SharpRazor** [comment]
## RHEL8
### 1. attach
```
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: Enter RHEL8 attach script with parameters: FCP list:1b0c 1d0d, INPUT WWPNS:0x5005076306031388 0x5005076306035388 0x5005076306039388 0x500507630603d388 0x5005076306081388 0x5005076306085388 0x5005076306089388 0x500507630608d388 0x5005076306101388 0x5005076306105388 0x5005076306109388 0x500507630610d388 0x5005076306131388 0x5005076306135388 0x5005076306139388 0x500507630613d388 0x5005076306181388 0x5005076306185388 0x5005076306189388 0x500507630618d388 0x50050763061b1388 0x50050763061b5388 0x50050763061b9388 0x50050763061bd388, LUN:0x4014403700000000, target_filename:vdb
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: Got FCP 1b0c
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: Got FCP 1d0d
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: Begin to enable zfcp mode ...
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: zfcp mode is already enabled
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: Begin to online FCP 1b0c ...
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: FCP 1b0c is online now
Dec 23 05:13:10 rhel85-eckd zvmguestconfigure[22525]: Begin to online FCP 1d0d ...
Dec 23 05:13:18 rhel85-eckd zvmguestconfigure[22525]: FCP 1d0d is online now
Dec 23 05:13:18 rhel85-eckd zvmguestconfigure[22525]: Port state of fcp: 1b0c is: Online
Dec 23 05:13:18 rhel85-eckd zvmguestconfigure[22525]: Port state of fcp: 1d0d is: Online
Dec 23 05:13:18 rhel85-eckd zvmguestconfigure[22525]: Scan on host host0 triggered
Dec 23 05:13:22 rhel85-eckd zvmguestconfigure[22525]: Scan on host host0 completed
Dec 23 05:13:22 rhel85-eckd zvmguestconfigure[22525]: Scan on host host1 triggered
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Scan on host host1 completed
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: lszfcp -P output port info:
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x500507680b22bac7
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x500507680b22bac6
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x5005076306185388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x5005076306035388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x500507680d820027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x500507680b21bac7
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x500507680d760027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x500507680b21bac6
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x500507680d120027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x500507680d060027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x5005076306105388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x50050763061bd388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: WWPNs shown under /sys/bus/ccw/drivers/zfcp/0.0.1b0c/:
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x5005076306035388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x5005076306185388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680b22bac6
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680b22bac7
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x5005076306035388 matched with content in ouput of lszfcp
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x5005076306185388 matched with content in ouput of lszfcp
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: WWPNs shown under /sys/bus/ccw/drivers/zfcp/0.0.1d0d/:
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x5005076306105388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x50050763061bd388
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680b21bac6
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680b21bac7
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680d060027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680d120027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680d760027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0x500507680d820027
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x5005076306105388 matched with content in ouput of lszfcp
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x50050763061bd388 matched with content in ouput of lszfcp
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Got valid wwpns, list of key is: 1b0c 1d0d, list of value is: 0x5005076306035388 0x5005076306185388 0x5005076306105388 0x50050763061bd388 .
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Discover WWPNs: 0x5005076306035388 0x5005076306185388 for FCP device: 1b0c
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x5005076306035388 0x4014403700000000 written into zfcp.conf
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1b0c 0x5005076306185388 0x4014403700000000 written into zfcp.conf
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Start to execute udev settle
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Execution of udevadm settle was done
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Try to detect disk /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Disk /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000 detected
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Discover WWPNs: 0x5005076306105388 0x50050763061bd388 for FCP device: 1d0d
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x5005076306105388 0x4014403700000000 written into zfcp.conf
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: 0.0.1d0d 0x50050763061bd388 0x4014403700000000 written into zfcp.conf
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Start to execute udev settle
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Execution of udevadm settle was done
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: The storage device is ready and found disk path: /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000.
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Begin to enable dm-multipath mode ...
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: dm-multipath mode is already enabled
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: scsi_id command get WWID:36005076306ffd3880000000000001437 for device /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000.
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Begin to reload udev rules ...
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Undev rules reload done
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: Exit RHEL8 attach script
Dec 23 05:13:23 rhel85-eckd zvmguestconfigure[22525]: zvmguestconfigure has successfully processed the reader files with exit code: 0.
Dec 23 05:13:23 rhel85-eckd systemd[1]: zvmguestconfigure.service: Succeeded.
```

### 2. detach
```
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: Enter RHEL8 detach script with parameters: FCP list:1b0c 1d0d, INPUT WWPNS:0x5005076306031388 0x5005076306035388 0x5005076306039388 0x500507630603d388 0x5005076306081388 0x5005076306085388 0x5005076306089388 0x500507630608d388 0x5005076306101388 0x5005076306105388 0x5005076306109388 0x500507630610d388 0x5005076306131388 0x5005076306135388 0x5005076306139388 0x500507630613d388 0x5005076306181388 0x5005076306185388 0x5005076306189388 0x500507630618d388 0x50050763061b1388 0x50050763061b5388 0x50050763061b9388 0x50050763061bd388, LUN:0x4014403700000000, target_filename:vdb, is_last_volume: 1.
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: Port state of fcp: 1b0c is: Online
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: Port state of fcp: 1d0d is: Online
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: lszfcp -P output port info:
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1b0c 0x500507680b22bac7
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1b0c 0x500507680b22bac6
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1b0c 0x5005076306185388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1b0c 0x5005076306035388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x500507680d820027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x500507680b21bac7
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x500507680d760027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x500507680b21bac6
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x500507680d120027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x500507680d060027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x5005076306105388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x50050763061bd388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: WWPNs shown under /sys/bus/ccw/drivers/zfcp/0.0.1b0c/:
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x5005076306035388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x5005076306185388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680b22bac6
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680b22bac7
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1b0c 0x5005076306035388 matched with content in ouput of lszfcp
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1b0c 0x5005076306185388 matched with content in ouput of lszfcp
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: WWPNs shown under /sys/bus/ccw/drivers/zfcp/0.0.1d0d/:
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x5005076306105388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x50050763061bd388
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680b21bac6
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680b21bac7
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680d060027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680d120027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680d760027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0x500507680d820027
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x5005076306105388 matched with content in ouput of lszfcp
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: 0.0.1d0d 0x50050763061bd388 matched with content in ouput of lszfcp
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: Got valid wwpns, list of key is: 1b0c 1d0d, list of value is: 0x5005076306035388 0x5005076306185388 0x5005076306105388 0x50050763061bd388 .
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: To find active disk path of LUN 0x4014403700000000, got WWPNs: 0x5005076306035388 0x5005076306185388 belonging to FCP device: 1b0c.
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: Try to detect disk /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: Disk /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000 detected
Dec 23 05:14:00 rhel85-eckd zvmguestconfigure[22891]: Found active disk path: /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000 .
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: scsi_id command get WWID: 36005076306ffd3880000000000001437 for device: /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Begin to flush cache data on /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000 ...
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Flush cache data on /dev/disk/by-path/ccw-0.0.1b0c-zfcp-0x5005076306035388:0x4014403700000000 was done
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Got map name: mpatha for WWID: 36005076306ffd3880000000000001437
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Exit code for multipath -f: 0
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: To remove LUN 0x4014403700000000 from file system, got WWPNs: 0x5005076306035388 0x5005076306185388 belonging to FCP device 1b0c.
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Begin to deregister SCSI device 0.0.1b0c:0x5005076306035388:0x4014403700000000 ...
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]:    Warning: Auto LUN scan enabled - cannot permanently deconfigure LUN
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: zFCP LUN 0.0.1b0c:0x5005076306035388:0x4014403700000000 deconfigured
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Deregistraion of 0.0.1b0c:0x5005076306035388:0x4014403700000000 was done
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Begin to deregister SCSI device 0.0.1b0c:0x5005076306185388:0x4014403700000000 ...
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]:    Warning: Auto LUN scan enabled - cannot permanently deconfigure LUN
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: zFCP LUN 0.0.1b0c:0x5005076306185388:0x4014403700000000 deconfigured
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Deregistraion of 0.0.1b0c:0x5005076306185388:0x4014403700000000 was done
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: This is last volume, will offline fcp device 1b0c
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: FCP device 0.0.1b0c deconfigured
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: FCP device 1b0c was offline now
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: To remove WWPN 0x5005076306035388 for 1b0c in zfcp.conf
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: WWPN 0x5005076306035388 for 1b0c in zfcp.conf removed
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: To remove WWPN 0x5005076306185388 for 1b0c in zfcp.conf
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: WWPN 0x5005076306185388 for 1b0c in zfcp.conf removed
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: To remove LUN 0x4014403700000000 from file system, got WWPNs: 0x5005076306105388 0x50050763061bd388 belonging to FCP device 1d0d.
Dec 23 05:14:01 rhel85-eckd zvmguestconfigure[22891]: Begin to deregister SCSI device 0.0.1d0d:0x5005076306105388:0x4014403700000000 ...
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]:    Warning: Auto LUN scan enabled - cannot permanently deconfigure LUN
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: zFCP LUN 0.0.1d0d:0x5005076306105388:0x4014403700000000 deconfigured
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: Deregistraion of 0.0.1d0d:0x5005076306105388:0x4014403700000000 was done
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: Begin to deregister SCSI device 0.0.1d0d:0x50050763061bd388:0x4014403700000000 ...
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]:    Warning: Auto LUN scan enabled - cannot permanently deconfigure LUN
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: zFCP LUN 0.0.1d0d:0x50050763061bd388:0x4014403700000000 deconfigured
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: Deregistraion of 0.0.1d0d:0x50050763061bd388:0x4014403700000000 was done
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: This is last volume, will offline fcp device 1d0d
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: FCP device 0.0.1d0d deconfigured
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: FCP device 1d0d was offline now
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: To remove WWPN 0x5005076306105388 for 1d0d in zfcp.conf
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: WWPN 0x5005076306105388 for 1d0d in zfcp.conf removed
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: To remove WWPN 0x50050763061bd388 for 1d0d in zfcp.conf
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: WWPN 0x50050763061bd388 for 1d0d in zfcp.conf removed
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: Begin to reload udev rules ...
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: Undev rules reload done
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: Exit RHEL8 detach script
Dec 23 05:14:20 rhel85-eckd zvmguestconfigure[22891]: zvmguestconfigure has successfully processed the reader files with exit code: 0.
Dec 23 05:14:20 rhel85-eckd systemd[1]: zvmguestconfigure.service: Succeeded.
```

**Comment by SharpRazor** [comment]
## SLES15
### 1. attach
```
```
### 2. detach
```
```

**Comment by SharpRazor** [comment]
## UBUNTU
### 1. attach
```
```
### 2. detach
```
```

File: zvmsdk/volumeops/templates/rhel7_attach_volume.j2 [hunk 1]
```diff
@@ -21,19 +21,35 @@ if [[ $fcp_count -eq 0 ]]; then
...
+    echo "FCP $fcp is online now"
+done
+
+# Print info of each FCP
+for fcp in ${fcp_list[@]}
+do
+    # print the status of FCP port for debug
+    host=$(lszfcp | grep -i $fcp | awk -F' ' '{print $2}')
+    if [[ -n "$host" ]]; then
+        port_state=$(cat /sys/class/fc_host/$host/port_state)
```
Comment (dongyanyang) [inline zvmsdk/volumeops/templates/rhel7_attach_volume.j2:48]: before cat the file, better to first check the file exists.

**Review by dongyanyang — APPROVED** [review]
lgtm

**Pull request merged**